From 19accb5229503dac7833f03713b978cb7fc48762 Mon Sep 17 00:00:00 2001
From: Junxiao Shi <git@mail1.yoursunny.com>
Date: Mon, 2 Dec 2024 15:33:38 +0000
Subject: [PATCH 31/33] Enable --client_ip flag in playback mode

refs #234

Signed-off-by: Junxiao Shi <git@mail1.yoursunny.com>
---
 src/sockperf.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/sockperf.cpp b/src/sockperf.cpp
index d84b035..6649b9b 100644
--- a/src/sockperf.cpp
+++ b/src/sockperf.cpp
@@ -1316,6 +1316,16 @@ static int proc_mode_playback(int id, int argc, const char **argv) {
      * List of supported under-load options.
      */
     const AOPT_DESC self_opt_desc[] = {
+        { OPT_CLIENTPORT,
+          AOPT_ARG,
+          aopt_set_literal(0),
+          aopt_set_string("client_port"),
+          "Force the client side to bind to a specific port (default = 0). " },
+        { OPT_CLIENTADDR,
+          AOPT_ARG,
+          aopt_set_literal(0),
+          aopt_set_string("client_ip", "client_addr"),
+          "Force the client side to bind to a specific address in IPv4, IPv6, UNIX domain socket format (default = 0). " },
         { OPT_REPLY_EVERY,
           AOPT_ARG,
           aopt_set_literal(0),
@@ -1461,6 +1471,11 @@ static int proc_mode_playback(int id, int argc, const char **argv) {
         rc = SOCKPERF_ERR_BAD_ARGUMENT;
     }
 
+    if (!rc) {
+        // --tcp option must be processed before
+        rc = parse_client_bind_info(common_obj, self_obj);
+    }
+
     if (rc) {
         const char *help_str = NULL;
         char temp_buf[30];
-- 
2.43.0

